interfaces/inc/c_pal/file_util_linux.h

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+
+#ifndef FILE_UTIL_LINUX_H
+#define FILE_UTIL_LINUX_H
+
+#ifdef __cplusplus
+#include <cstdint>
+#else
+#include <stdbool.h>
+#include <stdint.h>
+#endif
+
+#include "umock_c/umock_c_prod.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef WIN32 
+#include "windows.h"
+#define ebs_create_file CreateFileA
+#define ebs_close_handle CloseHandle
+#else
```

Comment:
Typically in the PAL project, we have the following conventions:

- There is a header in the interfaces/inc/c_pal which has the mockable interface
- The implementation for win32 is in win32/src/ (along with tests and specs in the win32 devdoc and tests folders)
- The implementation for Linux is in linux/src (and there are tests and devdoc's as well).

While I understand the reasoning for this to look very much like the win32 API (simple drop-in replacement is great!) I think we should keep the same convention, so I propose we:

- rename this file to "file_util.h"
- Get rid of the renames to CreateFileA and CloseHandle under "ifdef WIN32" here
- Move "file_util_linux.md" to the linux/devdoc folder
- Implementation for windows in win32/src/file_util.c can just be pass-through functions which call CreateFileA, CloseHandle, etc. That will allow the callers of this to have mockable functions for unit tests.

I don't think this is too much additional refactoring and then the module looks a lot more like our other PAL modules.

I guess having the "ifdef WIN32" part to either include windows.h or define the necessary symbols (GENERIC_READ, GENERIC_WRITE, etc.) is probably the easiest so we can leave that here.

Reply:
I think that keeping this as a MOCKABLE_FUNCTION on both windows and linux will also make it a lot easier to unit test things that use this (like in bsdl) without having to do something difference in the BSDL tests for windows or linux.

Reply:
Along with this, typically the "interfaces/devdoc/*.md" file describes the interface in a generic way. Might be nice to just have that MD file which says something for each function, like: "ebs_create_file" is a PAL wrapper for the Win32 API CreateFileA (link to MSDN page).